cryptodoc/src/08_signatures.rst

```diff
@@ -656,3 +656,193 @@ operations.
+   - Botan's hint generation differs slightly from [Dilithium-R3]_. This is discussed in `Hint Generation`_.
+
+.. _pubkey_signature/dilithium/val:
+
+Signature Validation
+^^^^^^^^^^^^^^^^^^^^
+
+The signature validation follows the :math:`\mathsf{Verify}` algorithm of Figure 4 of [Dilithium-R3]_. It is
+implemented in the ``Dilithium_Verification_Operation`` class, which receives the public key via the constructor.
+Message bytes are given to the object via consecutive calls of ``Dilithium_Verification_Operation::update``.
+
+.. admonition:: Dilithium_Verification_Operation::is_valid_signature()
+
+   **Input:**
+
+   -  ``pk = (rho, t_1, tr)``: public key
+   -  ``matrix``: public key matrix :math:`\mathbf{A}` (corresponds to L. 27, Fig. 4, [Dilithium-R3]_)
+   -  ``mu``:  hash of ``tr`` and the message ``msg`` (corresponds to L. 28, Fig. 4, [Dilithium-R3]_)
+   -  ``sig = (z, h, c)``: the signature
+   -  ``m``: Dilithium mode providing parameters (``gamma1``, ``gamma2``, ``beta``, ``omega``) and symmetric functions
+
+   **Output:**
+
+   -  ``true``, if the signature for message ``msg`` is valid. ``false`` otherwise.
+
+   **Steps:**
+
+   1. Check that the signature has the appropriate length and extract its parameters. Return ``false`` if
+      the signature length is invalid, ``z`` is no valid signature vector (i.e., ``z.polyvec_chknorm(gamma1 - beta)``), or
+      ``h`` is no valid hint vector (i.e., ``amount of 1's in h > omega``) (first and third check of L. 31, Fig. 4, [Dilithium-R3]_)
+   2. ``cp = Polynomial::poly_challenge(c)`` (L. 29, Fig. 4, [Dilithium-R3]_)
+   3. ``w1 = A*z - c*t*2^d`` (Second input of L. 30, Fig. 4, [Dilithium-R3]_)
+   4. ``w1 = PolynomialVector::polyvec_use_hint(h, w1, m)`` (L. 30, Fig. 4, [Dilithium-R3]_)
+   5. Signature is valid if ``c == H(mu || w1)`` (L. 31, Fig. 4, [Dilithium-R3]_)
+
+   **Notes:**
+
+   - ``matrix`` is already generated in NTT representation in the constructor via ``matrix = PolynomialMatrix::generate_matrix(rho, m)``.
+   - NTTs are performed as indicated by the comments in Fig. 4, [Dilithium-R3]_.
+   - mu = ``H(tr || msg)`` is already computed beforehand (in the constructor and using the ``update(msg)`` function).
+
+.. _pubkey_signature/dilithium/hint:
+
+Hint Generation
+^^^^^^^^^^^^^^^
+
+Dilithium uses a simple technique to reduce the size of the public key.
+Given the public matrix :math:`A` and :math:`t = As_1 + s_2`, the public key only contains the "low order" bits of :math:`t`.
+However, when computing with :math:`t`, a carry can occur, influencing the high order bits.
+In order to take care of this, Dilithium computes a "hint".
+The corresponding simple algorithm is :math:`\mathsf{MakeHint}_q` specified in Figure 3 of [Dilithium-R3]_.
+
+To see that Botan's hint computation on inputs ``(w0 - c*s2 + c*t0, w1)`` is equivalent to the specification of [Dilithium-R3]_, we look at the hint creation in Figure 3, L. 23 of [Dilithium-R3]_.
```

Comment:
But now I think we need to mention here that Botan's implementation dffers from Figure 3 in the spec and mention the Dilithium reference implementation as well.

Reply:
Done in [5382773](https://github.com/sehlen-bsi/botan-docs/pull/13/commits/53827733d53f5f188802f30135bf64bdbf3c1990)